gtsfm/evaluation/metrics.py

```diff
@@ -0,0 +1,287 @@
+        If the metric is a 1D distribution, the value is another dict which contains the summary of the data and also all the raw values (if they are stored).
+
+        Returns:
+            The metric as a dict representation explained above.
+        """
+        if self._dim == 0:
+            return {self._name: self._data.tolist()}
+
+        metric_dict = {SUMMARY_KEY: self.summary}
+        if self._data is not None:
+            metric_dict[DATA_KEY] = self._data.tolist()
+        return {self._name: metric_dict}
+
+    def save_to_json(self, json_filename):
+        """Saves this metric's dict representation to a JSON file.
+
+        Args:
+            Path to the json file.
+        """
+        io.save_json_file(json_filename, self.get_metric_as_dict())
+
+    @classmethod
+    def parse_from_dict(cls, metric_dict: Dict[str, Any]) -> GtsfmMetric:
+        """Creates a GtsfmMetric by parsing a dict representation.
+
+        It is assumed that the dict representation is the format created by GtsfmMetric.
+
+        Args:
+            metric_dict: Dict representation of the metric.
+
+        Returns:
+            Parsed GtsfmMetric instance.
+        """
+        if len(metric_dict) != 1:
+            raise AttributeError("Input metric dict should have a single key-value pair.")
+
+        metric_name = list(metric_dict.keys())[0]
+        metric_value = metric_dict[metric_name]
+
+        # 1D distribution metrics
+        if isinstance(metric_value, dict):
+            data = None
+            summary = None
+            if DATA_KEY in metric_value:
+                data = metric_value[DATA_KEY]
+            if SUMMARY_KEY in metric_value:
+                summary = metric_value[SUMMARY_KEY]
+            return cls(metric_name, data=data, summary=summary)
+
+        # Scalar metrics
+        return cls(metric_name, metric_value)
+
+
+class GtsfmMetricsGroup:
+    """Stores GtsfmMetrics from the same module. """
+
+    def __init__(self, name: str, metrics: List[GtsfmMetric]):
+        self._name = name
+        self._metrics = metrics
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def metrics(self):
+        return self._metrics
+
+    def add_metric(self, metric: GtsfmMetric):
+        self._metrics.append(metric)
+
+    def add_metrics(self, metrics: List[GtsfmMetric]):
+        self._metrics.extend(metrics)
+
+    def extend(self, metrics_group: GtsfmMetricsGroup):
+        self._metrics.extend(metrics_group.metrics)
+
+    def get_metrics_as_dict(self) -> Dict[str, Dict[str, Any]]:
+        metrics_dict = {}
+        for metric in self._metrics:
+            metrics_dict.update(metric.get_metric_as_dict())
+        return {self._name: metrics_dict}
+
+    def save_to_json(self, path: str):
+        io.save_json_file(path, self.get_metrics_as_dict())
+
+    @classmethod
+    def parse_from_dict(cls, metrics_group_dict) -> GtsfmMetricsGroup:
```

Comment:
Could you add A docstring explaining where/why this would be used in a GTSFM workflow? Parsing from json makes sense for later vis,  i I was confused about this  one

Reply:
this is only used for viz, I am adding documentation to the method, I dont think the documentation should say where its used though, thats upto the user.

Reply:
wouldn't we parse from JSON, not from a dict, in the viz code? or is it that we first read in from JSON as a dict, and then parse it?

Reply:
ok i looked and we call `parse_from_dict()` inside of `parse_from_json()`. But would we ever use `parse_from_dict()` besides from calling it inside the json classmethod? I guess i'm wondering if it's necessary

Reply:
I think its a useful abstraction. It would be a valid point if rewrote the parsing logic for both parse_from_json() and parse_from_dict(), but parse_from_json() is only calling parse_from_dict() right? 
I actually wanted to remove parse_from_json() and let the caller handle the IO, but this helper might reduce some json loading boilerplate.